include/gz/math/graph/Edge.hh

```diff
@@ -330,9 +333,19 @@ namespace graph
   };
 
   /// \brief An invalid directed edge.
+  // Deprecated in favor of NullEdge().
   template<typename E>
   DirectedEdge<E> DirectedEdge<E>::NullEdge(
     VertexId_P(kNullId, kNullId), E(), 1.0, kNullId);
+
+  /// \brief An invalid edge.
+  template<typename E, typename EdgeType>
+  EdgeType &NullEdge()
+  {
+    static EdgeType e(
```

Comment:
This is still problematic depending the given types `E`, and `EdgeType` because the destructor eventually runs when the program exits. According to the [Google style guide](https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables), it's best to allocate memory from the heap and never delete it, which will avoid calling the destructor. `gz::utils::NeverDestroyed` does exactly that, so I recommend we switch to that.

Reply:
trying `NeverDestroyed` in https://github.com/gazebosim/gz-math/pull/612/commits/3fa98549b1e781aca2f17909538c4891548963ad

Reply:
rerunning gz-sim CI [![Build Status](https://build.osrfoundation.org/buildStatus/icon?job=gz_sim-ci-pr_any-homebrew-amd64&build=693)](https://build.osrfoundation.org/job/gz_sim-ci-pr_any-homebrew-amd64/693/) https://build.osrfoundation.org/job/gz_sim-ci-pr_any-homebrew-amd64/693/